src/node-lib/curriculum-api-2023/queries/pupilLesson/pupilLesson.query.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  LessonContent,
+  LessonBrowseData,
+  lessonContentSchema,
+  lessonBrowseDataSchema,
+} from "./pupilLesson.schema";
+
+import errorReporter from "@/common-lib/error-reporter";
+import OakError from "@/errors/OakError";
+import { Sdk } from "@/node-lib/curriculum-api-2023/sdk";
+import {
+  InputMaybe,
+  Published_Mv_Synthetic_Unitvariant_Lessons_By_Year_6_0_0_Bool_Exp,
+} from "@/node-lib/curriculum-api-2023/generated/sdk";
+import keysToCamelCase from "@/utils/snakeCaseConverter";
+
+export const pupilLessonQuery =
+  (sdk: Sdk) =>
+  async (args: {
+    lessonSlug: string;
+    unitSlug?: string;
+    programmeSlug?: string;
+    isLegacy?: boolean;
+  }): Promise<{ content: LessonContent; browseData: LessonBrowseData }> => {
+    const { lessonSlug, unitSlug, programmeSlug, isLegacy } = args;
+
+    const browseDataWhere: InputMaybe<Published_Mv_Synthetic_Unitvariant_Lessons_By_Year_6_0_0_Bool_Exp> =
+      { lesson_slug: { _eq: lessonSlug } };
+
+    if (unitSlug) {
+      browseDataWhere["unit_slug"] = { _eq: unitSlug };
+    }
+
+    if (programmeSlug) {
+      browseDataWhere["programme_slug"] = { _eq: programmeSlug };
+    }
+
+    if (isLegacy !== undefined) {
+      browseDataWhere["is_legacy"] = { _eq: isLegacy };
+    }
+
+    const res = await sdk.pupilLesson({
+      browseDataWhere,
+      lessonSlug,
+    });
+
+    const [browseDataSnake] = res.browseData;
+
+    if (!browseDataSnake) {
+      throw new OakError({ code: "curriculum-api/not-found" });
+    }
+
+    if (browseDataSnake.lesson_data?.deprecated_fields?.is_sensitive) {
+      throw new OakError({ code: "curriculum-api/not-found" });
+    }
+
+    if (res.browseData.length > 1 && unitSlug && programmeSlug) {
+      const error = new OakError({
+        code: "curriculum-api/uniqueness-assumption-violated",
+      });
+      errorReporter("curriculum-api-2023::pupilLesson")(error, {
+        severity: "warning",
+        ...args,
+        res,
+      });
+    }
+
+    const [contentSnake] = res.content;
+
+    if (!contentSnake) {
+      throw new OakError({ code: "curriculum-api/not-found" });
+    }
+
+    if (res.content.length > 1) {
+      const error = new OakError({
+        code: "curriculum-api/uniqueness-assumption-violated",
+      });
+      errorReporter("curriculum-api-2023::pupilLesson")(error, {
+        severity: "warning",
+        ...args,
+        res,
+      });
+    }
+
+    lessonBrowseDataSchema.parse(browseDataSnake);
+    lessonContentSchema.parse(contentSnake);
+
+    // We've already parsed this data with Zod so we can safely cast it to the correct type
+    const browseData = keysToCamelCase(browseDataSnake) as LessonBrowseData;
```

Comment:
wait how can this be the same type with snake case keys and camel case keys?

Reply:
oh I see the `lessonBrowseDataSchema` is snake but the type is camel?

Reply:
browseDataSnake has a type derived from gql code gen so they aren't the same type.  Try hovering over and it should be apparent